lib/matplotlib/tests/test_pyplot.py

```diff
@@ -16,7 +17,19 @@ def test_pyplot_up_to_date():
     try:
         subprocess.run([sys.executable, str(gen_script)], check=True)
         new_contents = Path(plt.__file__).read_text()
-        assert orig_contents == new_contents
+
+        if orig_contents != new_contents:
+            diff_msg = '\n'.join(
+                difflib.unified_diff(
+                    orig_contents.split('\n'), new_contents.split('\n'),
+                    fromfile='found pyplot.py',
+                    tofile='expected pyplot.py',
+                    n=0, lineterm=''))
+            raise AssertionError(
```

Comment:
Can use either
```
assert success_cond, error_msg
```
or
```
if fail_cond: pytest.fail(error_msg)
```
either of which seem mildly preferable to throwing AssertionError explicitly.